### PR TITLE
[JENKINS-39225] Add French translations.

### PR DESCRIPTION
**Pull request by Nicolas-M**
# Description

See [JENKINS-39225](https://issues.jenkins-ci.org/browse/JENKINS-39225).
and https://issues.jenkins-ci.org/browse/JENKINS-40730

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [x] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees 

# Submitter Notes
* This change has no test because it only affects property files.
* To view the changes, simply launch Blue Ocean and go to `http://localhost:8080/jenkins/blue?language=fr`.

# Submitter Question
* I noticed that some components did not use strings from the message bundles. For example, the replay button has a constant value of '_Re-run_'. Assuming I need to create a new ticket for that, would someone be able to help me log in to JIRA? I can log in the Wiki without any problems...

_Edit_: I see that [JENKINS-40131](https://issues.jenkins-ci.org/browse/JENKINS-40131) had already been created for the 'Re-run' button. Nevertheless, other components such as the `<title>` of the status circles (e.g. "unstable") have not been translated yet.

### CODE REVIEW

**Comment by i386** [comment]
Merci beaucoup!

**Comment by michaelneale** [comment]
thanks @Nicolas-M, I will ask a french friend to review/proof read but this is great, much appreciated!

**Comment by Nicolas-M** [comment]
De rien! 

And sounds good. I fixed my JIRA login issues and created [JENKINS-40711](https://issues.jenkins-ci.org/browse/JENKINS-40711) for the missing strings I mentioned.

**Comment by michaelneale** [comment]
@Nicolas-M I added a new ticket (generally we try to track things with open tickets): https://issues.jenkins-ci.org/browse/JENKINS-40730 - just for release notes when this goes in. 

Thanks again!

**Comment by Nicolas-M** [comment]
@michaelneale Makes sense. I will keep that in mind in the future.

**Comment by michaelneale** [comment]
@Nicolas-M will merge this once get a review by french speakers. Can I make a joke about always being on vacation? Is that appropriate? ;)

File: blueocean-dashboard/src/main/resources/jenkins/plugins/blueocean/dashboard/Messages_fr.properties [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+EmptyState.pr=# Je pousse, tu tires\nLorsqu'une demande de tirage est ouverte pour le d\u00e9p\u00f4t _{0}_, Jenkins la testera et rapportera l'\u00e9tat de vos changements sur la demande de tirage sur GitHub.
+EmptyState.tests=Il n'y a aucune ex\u00e9cution de tests pour cette construction.\n\n
+Enable=Activer
+# home page / dashboard
+home.header.button.createpipeline=Nouveau Pipeline
+home.header.dashboard=Tableau de bord
+home.pipelineslist.header.branches=Branches
+home.pipelineslist.header.health=Sant\u00e9
+home.pipelineslist.header.name=Nom
+home.pipelineslist.header.pullrequests=Demandes de tirage
```
Comment (ndeloof) [inline blueocean-dashboard/src/main/resources/jenkins/plugins/blueocean/dashboard/Messages_fr.properties]: it sounds weird to me to translate "pull-request". I prefer such technical terms to be used in plain english.

File: blueocean-dashboard/src/main/resources/jenkins/plugins/blueocean/dashboard/Messages_fr.properties [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+pipelinedetail.activity.button.run=Lancer maintenant
+pipelinedetail.activity.header.branch=Branche
+pipelinedetail.activity.header.build=Construction
+pipelinedetail.activity.header.commit=Consigne
+pipelinedetail.activity.header.completed=Termin\u00e9
+pipelinedetail.activity.header.duration=Dur\u00e9e
+pipelinedetail.activity.header.message=Message
+pipelinedetail.activity.header.status=\u00c9tat
+pipelinedetail.branches.header.branch=Branche
+pipelinedetail.branches.header.commit=Consigne
```
Comment (ndeloof) [inline blueocean-dashboard/src/main/resources/jenkins/plugins/blueocean/dashboard/Messages_fr.properties]: Same comment. "Consigne" doesn't make any sense to me in a software development context, while all French people I know perfectly understand what a "commit" means, even if they don't know the actual meaning of this english noun.

File: blueocean-dashboard/src/main/resources/jenkins/plugins/blueocean/dashboard/Messages_fr.properties [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+pipelinedetail.branches.header.completed=Termin\u00e9
+pipelinedetail.branches.header.health=Sant\u00e9
+pipelinedetail.branches.header.message=Dernier message
+pipelinedetail.branches.header.status=\u00c9tat
+pipelinedetail.common.tab.activity=Activit\u00e9
+pipelinedetail.common.tab.branches=Branches
+pipelinedetail.common.tab.pullrequests=Demandes de tirage
```
Comment (ndeloof) [inline blueocean-dashboard/src/main/resources/jenkins/plugins/blueocean/dashboard/Messages_fr.properties]: idem

File: blueocean-dashboard/src/main/resources/jenkins/plugins/blueocean/dashboard/Messages_fr.properties [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+#run details
+rundetail.artifacts.button.download=T\u00e9l\u00e9charger l'artefact
+rundetail.artifacts.button.downloadAll.text=Tout t\u00e9l\u00e9charger
+rundetail.artifacts.button.downloadAll.title=T\u00e9l\u00e9charger tous les artefacts comme zip
+rundetail.artifacts.header.name=Nom
+rundetail.artifacts.header.size=Taille
+rundetail.artifacts.limit=Seuls les 100 premiers artefacts sont affich\u00e9s
+
+rundetail.changes.header.author=Auteur
+rundetail.changes.header.commit=Consigne
```
Comment (ndeloof) [inline blueocean-dashboard/src/main/resources/jenkins/plugins/blueocean/dashboard/Messages_fr.properties]: idem

File: blueocean-dashboard/src/main/resources/jenkins/plugins/blueocean/dashboard/Messages_fr.properties [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+rundetail.changes.header.date=Date
+rundetail.changes.header.message=Message
+rundetail.header.branch=Branche
+rundetail.header.changes.count={0} changements
+rundetail.header.changes.names=Changements par {0}
+rundetail.header.changes.none=Aucuns changements
+rundetail.header.commit=Consigne
```
Comment (ndeloof) [inline blueocean-dashboard/src/main/resources/jenkins/plugins/blueocean/dashboard/Messages_fr.properties]: idem

File: blueocean-dashboard/src/main/resources/jenkins/plugins/blueocean/dashboard/Messages_fr.properties [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+rundetail.header.tab.artifacts=Artefacts
+rundetail.header.tab.changes=Changements
+rundetail.header.tab.pipeline=Pipeline
+rundetail.header.tab.tests=Tests
+rundetail.pipeline.logs=Historique
```
Comment (ndeloof) [inline blueocean-dashboard/src/main/resources/jenkins/plugins/blueocean/dashboard/Messages_fr.properties]: Bad translation if you really want to provide one. I think everybody is fine with "logs"

**Comment by Nicolas-M** [comment]
@ndeloof Thanks for the comments. We use so many English terms when we talk software in French (here at least) that finding the proper translation is not always obvious.

File: blueocean-dashboard/src/main/resources/jenkins/plugins/blueocean/dashboard/Messages_fr.properties (hunk 2 above)
Comment (Nicolas-M) [inline blueocean-dashboard/src/main/resources/jenkins/plugins/blueocean/dashboard/Messages_fr.properties]: @ndeloof D'accord. Comment l'utiliseriez-vous dans une phrase par contre? Comme un verbe (i.e. "Commitez au dépôt"), ou bien comme un nom ("Faites un commit au dépôt")? Je ne suis pas certain comment je devrais corriger la traduction pour `EmptyState.activity`.

Comment (ndeloof) [inline blueocean-dashboard/src/main/resources/jenkins/plugins/blueocean/dashboard/Messages_fr.properties]: "_je commite, tu commites, il commite_" est ce que j'entends le plus souvent et qui semble ne pas perturber la compréhension. "_Faire un commit_" est sans doute plus propre pour limiter cet anglicisme.

**Comment by i386** [comment]
Thanks for the work here 💯

**Comment by guguxfr** [comment]
I've reviewed the translations, but not in the context of the software so I can't guarantee there is no non-sense.

I've noticed 

Hmm, on dirait qu'il n'y aucune
=> Hmm, on dirait qu'il n'y a aucune

Consignez au d\u00e9p\u00f4t _{0}_ ou lancer la construction manuellement.
=> Consignez au d\u00e9p\u00f4t _{0}_ ou lancez la construction manuellement.

et Jenkins commencera \u00e0 tester vos changements
=> I would have said "modifications" instead of "changements"

Essayez-vous et devenez le h\u00e9ro de votre \u00e9quipe.
=> Essayez et devenez le h\u00e9ros de votre \u00e9quipe.
or => Essayez-le et devenez le h\u00e9ros de votre \u00e9quipe.
or => Lancez-vous et devenez le h\u00e9ros de votre \u00e9quipe.

Les contructions de branches ne fonctionnent seulement avec
=> Les contructions de branches fonctionnent seulement avec
or => Les contructions de branches ne fonctionnent qu'avec

pipelinedetail.pullrequests.header.summary=Sommaire
=> isn't "Résumé" a better translation ?

rundetail.header.changes.none=Aucuns changements
=> "Aucun changement" feels more correct to me

Tous les tests ont pass\u00e9s avec succ\u00e8s.
=> Tous les tests sont pass\u00e9s avec succ\u00e8s.

The rest of the translations seem ok to me.

**Comment by Nicolas-M** [comment]
@guguxfr Thanks for the feedback. I agree with all of the changes that you proposed. The current translations have already been merged into master however. So, I've created #681. Let me know if you find anything else.
